src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs

```diff
@@ -234,7 +234,22 @@ internal QueryNode BindFunctionCall(FunctionCallToken functionCallToken)
 
             // If there isn't, bind as Uri function
             // Bind all arguments
-            List<QueryNode> argumentNodes = new List<QueryNode>(functionCallToken.Arguments.Select(ar => this.bindMethod(ar)));
+            List<QueryNode> argumentNodes = new List<QueryNode>(functionCallToken.Arguments.Select(argument =>

```

Comment:
It may be better to call `functionCallToken.Arguments.Select(...).ToList()` than `new List(...)`. If `Arguments` has a known collection size, then using `ToList()` could use an optimized path that avoids resizing. See: 
https://github.com/dotnet/runtime/issues/107560